cpp/tests/test_ad.cpp

```diff
@@ -91,12 +91,13 @@ TEST(Testad, ad_odeint)
     // The type of container used to hold the state vector.
     using value_type = ad_forward_t;
     using time_type  = value_type;
-    typedef std::vector<value_type> state_type;
+    const size_t dimension = 2;
+    typedef std::array<value_type, dimension> state_type; // 2-dimensional vector
```

Comment:
@hannemann-tamas I think this is not what @reneSchm meant. In line 81, you basically handover any type of state_type (which could be a one element vector) but then you access elems 0 and 1. @reneSchm Correct me if I am wrong and maybe directly post your suggestion.

Reply:
I suggested using `Eigen::Matrix<value_type, 2, 1>` as we usually use Eigen vector types in calculations, but the `std::array` has a fixed size as well, so it does fix the issue I had with the older version.

And since `state_type` is fixed and not a template, we cannot simply pass a vector.

Though, it would be nicer to use the `using` keyword instead of `typedef`.

Reply:
Wait, I did not see the change to harmonic_oscillator in the single commits. No, I only wanted to change the static type `state_type`, using templates here seems needlessly complicated.